### PR TITLE
add suggestions to composer

### PR DESCRIPTION
**Pull request by Rotzbua**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Doc update?   | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | no
| License       | MIT

If installed by composer, show required extensions for optional features.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,6 +22,12 @@
         "mockery/mockery": "~0.9.1",
         "symfony/phpunit-bridge": "~3.3@dev"
     },
+    "suggest": {
+        "ext-openssl": "required for encrypted connections, dkim, smime, NTLMAuthenticator, ...",
+        "ext-bcmath": "required for NTLMAuthenticator",
+        "ext-mbstring": "mbstring or iconv required for multi byte support",
```
Comment (fabpot) [inline composer.json]: I suggest that we require `symfony/polyfill-mbstring` instead.

File: CHANGES [hunk 2]
```diff
@@ -5,6 +5,7 @@ Changelog
 ------------------
 
  * n/a
```
Comment (fabpot) [inline CHANGES]: should be removed

File: composer.json [hunk 3]
```diff
@@ -22,6 +22,12 @@
...
+        "ext-iconv": "mbstring or iconv required for multi byte support, iconv is required for NTLMAuthenticator"
```
Comment (fabpot) [inline composer.json]: Same here, what about requiring `symfony/polyfill-iconv` (which is needed for the mbstring polyfill)?

**Review by fabpot — COMMENTED** [review]
/cc @nicolas-grekas

File: composer.json (hunk 1 above)
Comment (Rotzbua) [reply composer.json]: I guess a polyfill is not required.
I think there should be no real php7.x environment without extensions mbstring or iconv.
Also "suggest" only creates a notice if the extension is not installed and will be shown after `$ composer require "swiftmailer/swiftmailer:^6.0"`. I only added the "suggestions" to create a hint for misconfigured environments while using composer and not while runtime.

Comment (fabpot) [reply composer.json]: For the record, I think that "suggest" are useless. Nobody reads them. And it's a one time thing.

**Comment by fabpot** [comment]
Thanks for the PR. As announced some time ago, Swiftmailer is not maintained anymore.

[Symfony Mailer](https://symfony.com/doc/current/mailer.html) is the next evolution of Swiftmailer.
It provides the same features with support for modern PHP code and support for third-party providers.

**Pull request closed without merging**